src/test/logging_tests.cpp

```diff
@@ -33,4 +32,108 @@ BOOST_AUTO_TEST_CASE(logging_timer)
+    LogInstance().m_log_threadnames = false;
+
+    // A string containing 1023 bytes.
+    std::string s1023(1023, 'a');
+
+    SetMockTime(std::chrono::seconds{1});
+
+    size_t prev_log_file_size, curr_log_file_size;
+    GetLogFileSize(prev_log_file_size);
+
+    // Log 1 MiB, this should be allowed.
+    for (int i = 0; i < 1024; ++i) {
+        LogFromFixedLocation(s1023);
+    }
+    GetLogFileSize(curr_log_file_size);
+    BOOST_CHECK(curr_log_file_size - prev_log_file_size == 1024 * 1024);
+
+    LogFromFixedLocation("This should trigger rate limiting");
+    GetLogFileSize(prev_log_file_size);
+
+    // Log 0.5 MiB, this should not be allowed and all messages should be dropped.
+    for (int i = 0; i < 512; ++i) {
+        LogFromFixedLocation(s1023);
+    }
+    GetLogFileSize(curr_log_file_size);
+    BOOST_CHECK(curr_log_file_size - prev_log_file_size == 0);
+
+    // Let one hour pass.
+    SetMockTime(std::chrono::seconds{60 * 60 + 1});
+    LogFromFixedLocation("This should trigger the quota usage reset");
+    GetLogFileSize(prev_log_file_size);
+
+    // Log 1 MiB, this should be allowed since the usage was reset.
+    for (int i = 0; i < 1024; ++i) {
+        LogFromFixedLocation(s1023);
+    }
+    GetLogFileSize(curr_log_file_size);
+    BOOST_CHECK(curr_log_file_size - prev_log_file_size == 1024 * 1024);
+
+    LogFromFixedLocation("This should trigger rate limiting");
+    GetLogFileSize(prev_log_file_size);
+
+    // Log 1 MiB, this should not be allowed and all messages should be dropped.
+    for (int i = 0; i < 1024; ++i) {
+        LogFromFixedLocation(s1023);
+    }
+    GetLogFileSize(curr_log_file_size);
+    BOOST_CHECK(curr_log_file_size - prev_log_file_size == 0);
+
+    LogFromFixedLocation("This should also be dropped and bring the total of dropped bytes above 1 MiB.");
+    GetLogFileSize(curr_log_file_size);
+    BOOST_CHECK(curr_log_file_size - prev_log_file_size == 0);
+
+    // Let another hour pass
+    SetMockTime(std::chrono::seconds{2 * (60 * 60 + 1)});
+    LogFromFixedLocation("Normally this would reset the usage but it does not because we dropped more than 1 MiB in the previous hour.");
+    GetLogFileSize(prev_log_file_size);
+
+    // Log 1 MiB, this should not be allowed and all messages should be dropped.
+    for (int i = 0; i < 1024; ++i) {
+        LogFromFixedLocation(s1023);
+    }
+    GetLogFileSize(curr_log_file_size);
+    BOOST_CHECK(curr_log_file_size - prev_log_file_size == 0);
+
+    // Let another hour pass
+    SetMockTime(std::chrono::seconds{3 * (60 * 60 + 1)});
+    LogFromFixedLocation("This should trigger the quota usage reset");
+    GetLogFileSize(prev_log_file_size);
+
+    // Log 1 MiB, this should be allowed since the usage was reset.
+    for (int i = 0; i < 1024; ++i) {
+        LogFromFixedLocation(s1023);
+    }
+    GetLogFileSize(curr_log_file_size);
+    BOOST_CHECK(curr_log_file_size - prev_log_file_size == 1024 * 1024);
```

Comment:
Both windows CI builds are failing on this line. See [cirrus](https://cirrus-ci.com/task/6258137339527168?logs=ci#L642).